enos/modules/docker_minio/main.tf

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+terraform {
+  required_providers {
+    docker = {
+      source  = "kreuzwerker/docker"
+      version = "3.0.1"
+    }
+
+    enos = {
+      source = "app.terraform.io/hashicorp-qti/enos"
+    }
+  }
+}
+
+variable "image_name_server" {
+  description = "Name of Docker Image for minio server"
+  type        = string
+  default     = "docker.mirror.hashicorp.services/minio/minio:latest"
+}
+variable "image_name_client" {
+  description = "Name of Docker Image for minio client"
+  type        = string
+  default     = "docker.mirror.hashicorp.services/minio/mc:latest"
+}
+variable "network_name" {
+  description = "Name of Docker Networks to join"
+  type        = list(string)
+}
+variable "container_name" {
+  description = "Name of Docker Container"
+  type        = string
+  default     = "minio"
+}
+variable "region" {
+  description = "AWS Region"
+  type        = string
+  default     = "us-east-1"
+}
+variable "bucket_name" {
+  description = "Name of storage bucket"
+  type        = string
+  default     = "testbucket" # this needs to match the bucket in policy.json
+}
+variable "root_user" {
+  description = "Username for minio root user"
+  type        = string
+  default     = "minio"
+}
+variable "root_password" {
+  description = "Password for minio root user"
+  type        = string
+  default     = "minioadmin"
+}
+variable "user_id" {
+  description = "Username/Access Key Id for user that can access bucket"
+  type        = string
+  default     = "testuser"
+}
+variable "user_password" {
+  description = "Password/Secret Access Key for user that can access bucket"
+  type        = string
+  default     = "password"
+}
+variable "user_access_key_id" {
+  description = "Access Key Id for user that can access bucket"
+  type        = string
+  default     = "useraccesskeyid"
+}
+variable "user_secret_access_key" {
+  description = "Secret Access Key for user that can access bucket"
+  type        = string
+  default     = "secretaccesskey"
```

Comment:
Is this no-separators-naming style some kind of minio convention?

Reply:
What are you referring to here? The default values? It can be set to anything (as long it as it meets some requirements). If you were to minio generate these values automatically, it would look something like
```
Access Key: 2B7G5EMQAANJX6YQ0H0I
Secret Key: kgd9tH29fx+ymWWDBAZss5KcMxqAa3iZXEw7lxWi
```

Reply:
The default is `secretaccesskey`. I was wandering why it's not `secret-access-key` for example.

Reply:
It can be set to anything as long as it's greater than 8 characters.